### PR TITLE
dailymotion shortcode: add support for dailymotion widget http://www.dailymotion.com/widgets

### PR DESCRIPTION
**Pull request by MathildeS**
adding support for dailymotion widget which is available for dailymotion partners at http://www.dailymotion.com/widgets
#### Changes proposed in this Pull Request:
- Shortcode [dailymotion-widget id=ID]
- Widget embed reversal
- Corresponding unit tests 
#### Testing instructions:
- you can test by passing this widget id: 574dadf2a94fba04f31ec1de 
- for the embed reversal, the code is as follow: 
  `<div class="dailymotion-widget" data-placement="574dadf2a94fba04f31ec1de"></div><script>(function(w,d,s,u,n,e,c){w.PXLObject = n; w[n] = w[n] || function(){(w[n].q = w[n].q || []).push(arguments);};w[n].l = 1 * new Date();e = d.createElement(s); e.async = 1; e.src = u;c = d.getElementsByTagName(s)[0]; c.parentNode.insertBefore(e,c);})(window, document, "script", "//api.dmcdn.net/pxl/client.js", "pxl");</script>`

### CODE REVIEW

**Comment by MathildeS** [comment]
this PR is ready for review

**Comment by lezama** [comment]
> this PR is ready for review

Cool, @eliorivero are you on it?

File: modules/shortcodes/dailymotion.php [hunk 1]
```diff
@@ -327,3 +327,61 @@ function jetpack_dailymotion_embed_reversal( $content ) {
...
+ * @return string html
+ */
+function jetpack_dailymotion_widget_shortcode( $atts ) {
+	if ( isset( $atts['id'] ) ) {
+		$id = $atts['id'];
+	} else {
+		return '<!--Dailymotion error: bad or missing ID-->';
+	}
+
+	$output="<div class='dailymotion-widget' data-placement='" . $id . "'></div>";
```
Comment (eliorivero) [inline modules/shortcodes/dailymotion.php]: needs spacing around the equal sign. Also, if you're using " you can just write:
```php
$output = "<div class='dailymotion-widget' data-placement='$id'></div>";
```

File: modules/shortcodes/dailymotion.php [hunk 2]
```diff
@@ -327,3 +327,61 @@ function jetpack_dailymotion_embed_reversal( $content ) {
...
+}
+
+add_shortcode( 'dailymotion-widget', 'jetpack_dailymotion_widget_shortcode' );
+
+/**
+ * Embed Reversal for Dailymotion widget
+ */
+function jetpack_dailymotion_widget_reversal( $content ) {
+
+	if ( false === stripos( $content, 'dailymotion-widget' ) ) {
```
Comment (jeherve) [inline modules/shortcodes/dailymotion.php]: I'd recommend using `has_shortcode()` here:
https://codex.wordpress.org/Function_Reference/has_shortcode

File: modules/shortcodes/dailymotion.php [hunk 3]
```diff
@@ -327,3 +327,61 @@ function jetpack_dailymotion_embed_reversal( $content ) {
 }
 
 add_filter( 'pre_kses', 'jetpack_dailymotion_embed_reversal' );
+
+/**
+ * DailyMotion Widget Shortcode
+ *
+ * Example:
+ * [dailymotion-widget id=XYZ]
+ *
```
Comment (eliorivero) [inline modules/shortcodes/dailymotion.php]: please add the following to all the phpdocs in this file:
```php
* @module shortcodes
```
For reference, [see this](https://github.com/Automattic/jetpack/blob/master/modules/shortcodes/vimeo.php#L115).

File: modules/shortcodes/dailymotion.php [hunk 4]
```diff
@@ -327,3 +327,61 @@ function jetpack_dailymotion_embed_reversal( $content ) {
...
+	$output="<div class='dailymotion-widget' data-placement='" . $id . "'></div>";
+	$output .= '<script>(function(w,d,s,u,n,e,c){w.PXLObject = n; w[n] = w[n] || function(){(w[n].q = w[n].q || []).push(arguments);};w[n].l = 1 * new Date();e = d.createElement(s); e.async = 1; e.src = u;c = d.getElementsByTagName(s)[0]; c.parentNode.insertBefore(e,c);})(window, document, "script", "//api.dmcdn.net/pxl/client.js", "pxl");</script>';
+
+	return $output;
+}
+
+add_shortcode( 'dailymotion-widget', 'jetpack_dailymotion_widget_shortcode' );
+
+/**
+ * Embed Reversal for Dailymotion widget
```
Comment (eliorivero) [inline modules/shortcodes/dailymotion.php]: missing `@param string $content`

File: modules/shortcodes/dailymotion.php (hunk 3 above)
Comment (MathildeS) [inline modules/shortcodes/dailymotion.php]: does this commit match what you expected @eliorivero ? : https://github.com/Automattic/jetpack/pull/4550/commits/926368d878cfd1ebb99ab020cd01e8acfe983343 ?

File: modules/shortcodes/dailymotion.php (hunk 4 above)
Comment (MathildeS) [inline modules/shortcodes/dailymotion.php]: see https://github.com/Automattic/jetpack/pull/4550/commits/e925b0d9bdae84589c311c0f930cf767ed0646f4 
following your advice, I added @return and @param annotations on all the phpdocs where they were missing in this file

File: modules/shortcodes/dailymotion.php (hunk 2 above)
Comment (MathildeS) [inline modules/shortcodes/dailymotion.php]: @jeherve  are you sure this is what we wanted to do? I'm not looking for the [dailymotion-widget] shortcode but for the simple "dailymotion-widget" string (that appears in `<div class="dailymotion-widget" data-placement="XXXXXXXXX">`). 

I thought that this was the way to do as this is what is done in other embed reversals, for instance:
- `function dailymotion_channel_reversal( $content ) {
	if ( false === stripos( $content, 'dailymotion.com/badge/' ) ) {
		return $content;
	}`
- `function dailymotion_embed_to_shortcode( $content ) {
	if ( false === stripos( $content, 'www.dailymotion.com/swf/' ) ) {
		return $content;
	}`

**Comment by MathildeS** [comment]
also, should I add the  `@since 4.3.1` mention in the phpdocs in the two functions I added in dailymotion.php file? I don't see this kind of annotation anywhere else in the file. 

I guess I have to change it in the unit tests as well (it is currently set to `@since 4.2.0`).

File: modules/shortcodes/dailymotion.php (hunk 3 above)
Comment (eliorivero) [inline modules/shortcodes/dailymotion.php]: Yes, perfect 👍

File: modules/shortcodes/dailymotion.php (hunk 2 above)
Comment (jeherve) [inline modules/shortcodes/dailymotion.php]: Good call, my bad 👍

**Comment by MathildeS** [comment]
is this PR ok? do I need to add something? maybe some change about the @since version ? (see my comment above)

**Comment by jeherve** [comment]
@MathildeS We haven't reviewed your PR yet, but we'll take care of it once Jetpack 4.3 is released. If everything goes well, your code will then be included in the release that follows.

You will indeed need to change the `@since` version at some point, but I'd recommend waiting until we review your PR to do that, as we might end up doing point releases right after 4.3, before to actually review your code.

**Comment by rcoll** [comment]
@MathildeS This looks pretty good to me and I can confirm the shortcode is working as expected. With that said, I was unable to get the reversal to work as expected. Have you tested this with the latest permutation of their embed code? It looks like it may have changed a bit, but I might also be doing something wrong.

**Comment by samhotchkiss** [comment]
Closing for now due to inactivity.

**Pull request closed without merging**